includes/admin/class-wcs-admin-empty-list-content-manager.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Class WCS_Admin_Empty_List_Content_Manager
+ *
+ * @package WooCommerce Subscriptions
+ * @since 6.2.0
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * A class for managing the content displayed in the WooCommerce → Subscriptions admin list table when no results are found.
+ */
+class WCS_Admin_Empty_List_Content_Manager {
+
+	/**
+	 * Initializes the class and attach callbacks.
+	 */
+	public static function init() {
+		add_action( 'admin_enqueue_scripts', array( __CLASS__, 'enqueue_scripts_and_styles' ) );
+	}
+
+	/**
+	 * Gets the content to display in the WooCommerce → Subscriptions admin list table when no results are found.
+	 *
+	 * @return string The HTML content for the empty state if no subscriptions exist, otherwise a string indicating no results.
+	 */
+	public static function get_content() {
+		$content = __( 'No subscriptions found.', 'woocommerce-subscriptions' );
+
+		if ( self::should_display_empty_state() ) {
+			$content = wc_get_template_html(
+				'html-admin-empty-list-table.php',
+				[],
+				'',
+				WC_Subscriptions_Core_Plugin::instance()->get_subscriptions_core_directory( 'templates/admin/' )
+			);
+		}
+
+		// Backwards compatibility for the woocommerce_subscriptions_not_found_label filter.
+		if ( has_action( 'woocommerce_subscriptions_not_found_label' ) ) {
```

Comment:
As you recently found, we could use the following code here:

```
$content = apply_filters_deprecated( 'woocommerce_subscriptions_not_found_label', [ $content ], '6.2.0', 'woocommerce_subscriptions_not_found_content' );
```

But I think the current code is fine and consistent with what we've done in other areas